rpc/auth/email/provider.go

```diff
@@ -170,7 +170,8 @@ func (p *AuthProvider) extractVerifier(verifier string) (emailAddress string, se
 	if len(parts) != 2 {
 		return "", "", fmt.Errorf("invalid verifier")
 	}
-	return parts[0], parts[1], nil
+	emailAddress = strings.TrimSpace(strings.ToLower(parts[0]))
+	return emailAddress, parts[1], nil
```

Comment:
Can we add a top-level function called `NormalizeEmail(email string) string`, which would be called from this (or parent) function?

Rationale: This is important logic & it feels a bit hidden inside of the `extractVerifier()` method. Perhaps the new function would have some extra unit tests too and could be expanded over time.